### DISCUSSION TITLE
Add Missable Achievement Notification

### DISCUSSION THREAD

**Discussion by eldexterr** [RetroAchievements/RAIntegration#756] [Ideas]
We got Measured (very cool feature), I was thinking, what about a way to make a notification pop-up when a missable achievement can be triggered?

The notification can use a diferent color (like the mastery one) so the player doens't think he just earned an achievement, also an option to disable this notifications, maybe they're players who doesn't want to get kind of spoiled by the notifications

**Comment by Sanaki** [comment]
I actually was just coming here to suggest this same thing, and then some. To be specific, I'd like to see support added for a secondary "achievement" type that's purely informational. Preferably that could trigger multiple times in a single session, similar to a leaderboard, but I don't know how viable that is. I'd actually put missables as a subcategory of the information type, primarily for the purpose of being able to add toggles to the achievements menu to disable all information popups (for those who may not want to see them at all), and further being able to leave them enabled but still disable missable notices for players who dislike seeing that sort of "spoiler".

The inspiration for this was [this achievement](https://retroachievements.org/achievement/82088), which pops on the title screen just to deliver information.

**Comment by Keltron3030** [comment]
Seems that you could use a leaderboard for that function right now.

**Comment by Sanaki** [comment]
Leaderboards would work in place of the achievement I linked, but not being able to toggle that specific type off would be problematic for the interest level I was hearing in discord chat. Some people want to use them for frequent info popups, which should be something a player can easily opt out of without losing functionality. Same goes for missables, since some players don't want to know about them.

**Comment by Jamiras** [comment]
The Trigger flag (#681) provides visual feedback when certain challenges are available. I believe it can reasonably be used to indicate when a missable achievement could be earned.

**Reply by Sanaki** [reply]
Trigger should be sufficient for this purpose, yes. The information display method I mentioned further up still has potential uses, though not as many now.

**Reply by FluffyEmmy** [reply]
This was just brought up in discord by Themooncow. 

A "missed" category on the achievement list separation would be useful, such as "active challenges" for trigger.

**Reply by Jamiras** [reply]
Having read through the "proposal" by Themooncow, the basic idea is to add _new_ logic to every Trigger achievement that indicates when the user has gotten past the point where the achievement is no longer earnable so they can be listed in the overlay as Missed instead of Locked.

I'm not fond of this idea as it requires adding additional logic. Trigger works nicely with the existing logic. You just flag the final condition(s) that cause the achievement to trigger, and when all the other conditions are "primed", the indicator is shown. 

I'm also not sure how it would affect the UI. The category would either appear before the Locked list and make it more difficult to see what might normally be the next thing to be unlocked, or after the Locked list in which case it would never be seen.

I can think of a few games that show challenge indicators during a level/mission which get crossed out, greyed out, or hidden once the challenge has been failed, but those are for earning stars or bonus rewards for the level, not for earning achievements. And Trigger mimics that behavior - display for however long the challenge is applicable, then hide the display.

Additionally, the "missed" logic may not have anything to do with the achievement itself. Here's a recent missable that I had to figure out: <https://retroachievements.org/achievement/162493>. The achievement title and description provide very little information on how (or when) it can be earned. The way to unlock it is effectively laid out in the comments - speak to a specific NPC during a specific period of the game. There's no Trigger indicator because it's not a challenge and you wouldn't want the Trigger indicator to be on screen for the entire duration that NPC could be spoken to. And even if you did get a popup saying the achievement could no longer be earned when you defeat some unrelated boss, would it really help you, or just be confusing?

Yes, it sucks to miss a missable, but name me one other achievement platform that even specifies which achievements are missable, let alone tries to provide any sort of indicator when they're available to be earned. I think the underlying problem is that missables prevent earning the mastery badge, and people are very vocal when they have to put additional unplanned effort into unlocking the mastery badge.

**Reply by Sanaki** [reply]
From my perspective, the only thing we still need missable-wise is a simple checkbox on the achievement that allows the site/frontend to display an icon next to it if the user has that enabled. That's not a logic feature though. For in-game missable stuff, trigger is more than sufficient.

**Reply by televandalist** [reply]
Yep. I've said many times that I would have no issues marking missables if we could toggle them from view as players. 



> The Trigger flag (#681) provides visual feedback when certain challenges are available. I believe it can reasonably be used to indicate when a missable achievement could be earned.

I experimented with this a bit in Trails in the Sky with missable sidequests. The notifications would appear when the player would be close to the point of no return and then disappear when it was too late. It worked really well, but I ended up scrapping it since it couldn't be turned on/off by players.

**Reply by slashtangent** [reply]
What about a setting that the player can set to get notified if trigger is reset? As is it's very easy to not notice that the trigger condition reset.

**Reply by Jamiras** [reply]
> It worked really well, but I ended up scrapping it since it couldn't be turned on/off by players.

They're labelled as "Challenge Indicators" and _can_ be turned off, both in the DLL (via overlay settings) and RetroArch (settings > achievements).

![image](https://user-images.githubusercontent.com/32680403/135189557-4c7fd9fa-f0c9-4e54-bfc9-6f313b670b6a.png)

![image](https://user-images.githubusercontent.com/32680403/135189511-3ea6c657-3d14-400f-9ae5-ec298ed3571e.png)

**Reply by televandalist** [reply]
What!? 😭
